scripts/west_commands/packages.py

```diff
@@ -0,0 +1,152 @@
+            "pip",
+            help="manage pip packages",
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            epilog=textwrap.dedent(
+                """
+            Manage pip packages:
+
+                Run 'west packages pip' to print all pip packages or requirements
+                files needed by Zephyr and the active modules.
+
+                The output is compatible with the requirements file format itself.
+            """
+            ),
+        )
+
+        pip_parser.add_argument(
+            "--install",
+            action="store_true",
+            help="Install pip packages instead of listing them. "
+            "A single 'pip install' command is built and executed. "
+            "Additional pip arguments can be passed after a -- separator "
+            "from the original 'west packages pip --install' command. For example pass "
+            "'--dry-run' to pip not to actually install anything, but print what would be.",
+        )
+
+        return parser
+
+    def do_run(self, args, unknown):
+        if len(unknown) > 0 and unknown[0] != "--":
+            self.die(
+                f'Unknown argument "{unknown[0]}"; '
+                'arguments for the manager should be passed after "--"'
+            )
+
+        if args.manager == "pip":
+            return self.do_run_pip(args, unknown[1:])
+
+        # Unreachable but print an error message if an implementation is missing.
+        self.die(f'Unsupported package manager: "{args.manager}"')
+
+    def do_run_pip(self, args, manager_args):
+        packages = []
+        requirements = []
+
+        if not args.modules or "zephyr" in args.modules:
+            requirements.append(ZEPHYR_BASE / "scripts/requirements.txt")
+
+        for module in zephyr_module.parse_modules(ZEPHYR_BASE, self.manifest):
+            module_name = module.meta.get("name", None)
+            if args.modules and module_name not in args.modules:
+                continue
+
+            # Get the optional pip section from the package managers
+            pip = module.meta.get("package-managers", {}).get("pip")
+            if pip is None:
+                continue
+
+            # Add individual listed packages
+            packages += pip.get("packages", [])
+            # Add requirements files
+            requirements += [Path(module.project) / r for r in pip.get("requirements", [])]
+
+        if args.install:
+            if not in_venv():
+                self.die("Running pip install outside of a virtual environment")
+
+            if len(packages) > 0 or len(requirements) > 0:
+                subprocess.check_call(
+                    [sys.executable, "-m", "pip", "install"]
+                    + packages
+                    + list(chain.from_iterable([("-r", r) for r in requirements]))
+                    + manager_args
+                )
+            else:
+                self.inf("Nothing to install")
```

Comment:
better, but `Nothing to install` is misleading when it's because the module does not exists.

For example, existing module without a `packages` section in the `module.yml` :
```
$ west packages --module hal_st pip --install
Nothing to install
```

a non-existing module, in this case user should have been informed that the module was invalid / unknown.
It could be the user was using invalid name, like `hal_ti` instead of `ti`, or misspelled the name.

```
$ west packages --module hal_ti pip --install
Nothing to install
```

and a valid module with a package section:
```
$ west packages --module ti pip --install
Requirement already satisfied: Jinja2 in /home/tora/zephyr-venv/lib/python3.10/site-packages (from -r /projects/github/ncs/modules/hal/ti/scripts/requirements.txt (line 1)) (3.1.4)
```

Especially if user combines a valid module together with a typo, then user may believe everything installs correctly, like in this case where two modules are given, invalid `typy` and correct `ti`:
```
$  west packages --module typy --module ti pip --install
Requirement already satisfied: Jinja2 in /home/tora/zephyr-venv/lib/python3.10/site-packages (from -r /projects/github/ncs/modules/hal/ti/scripts/requirements.txt (line 1)) (3.1.4)
```

Reply:
Added a check for unknown passed module names.

Reply:
Looks good.

Small observation, feel free to decide if this should have a changed behavior (perhaps current behavior is intentional). 

Calling with a module which doesn't define package manager correctly results in:
```
$ west packages --module hal_st pip --install
Nothing to install
```

but passing along an extra module (technically Zephyr itself is not a module) results in everything install and no feedback that one module did not specify any package manager:
```
$ west packages --module zephyr --module hal_st pip --install
Ignoring windows-curses: markers 'sys_platform == "win32"' don't match your environment
Ignoring python-magic-bin: markers 'sys_platform == "win32"' don't match your environment
Requirement already satisfied: pyelftools>=0.29 in /home/tora/zephyr-venv/lib/python3.10/site-packages (from -r /projects/github/ncs/zephyr/scripts/requirements-base.txt (line 7)) (0.31)
...
```